src/ui/components/Video/NodeHighlighter/PreviewNodeHighlighter.tsx

```diff
@@ -0,0 +1,225 @@
+
+function nodeNeedsHighlighting(quads: BoxModelQuads) {
+  return quads.margin.length || quads.border.length || quads.padding.length || quads.content.length;
+}
+
+interface Point {
+  x: number;
+  y: number;
+}
+
+interface QuadWithBounds {
+  p1: Point;
+  p2: Point;
+  p3: Point;
+  p4: Point;
+  bounds: {
+    bottom: number;
+    height: number;
+    left: number;
+    right: number;
+    top: number;
+    width: number;
+    x: number;
+    y: number;
+  };
+}
+
+function buildBoxQuads(array: Quads) {
+  assert(array.length % 8 == 0, "quads length must be a multiple of 8");
+  array = [...array];
+  const rv = [];
+  while (array.length) {
+    const [x1, y1, x2, y2, x3, y3, x4, y4] = array.splice(0, 8);
+    rv.push(
+      DOMQuad.fromQuad({
+        p1: { x: x1, y: y1 },
+        p2: { x: x2, y: y2 },
+        p3: { x: x3, y: y3 },
+        p4: { x: x4, y: y4 },
+      })
+    );
+  }
+  return rv;
+}
+
+function getOuterQuad(region: BoxModelKeys, boxModelQuads: BoxModelQuads): QuadWithBounds | null {
+  const quads = boxModelQuads[region];
+  if (!quads || !quads.length) {
+    return null;
+  }
+
+  const quad = {
+    p1: { x: Infinity, y: Infinity },
+    p2: { x: -Infinity, y: Infinity },
+    p3: { x: -Infinity, y: -Infinity },
+    p4: { x: Infinity, y: -Infinity },
+    bounds: {
+      bottom: -Infinity,
+      height: 0,
+      left: Infinity,
+      right: -Infinity,
+      top: Infinity,
+      width: 0,
+      x: 0,
+      y: 0,
+    },
+  };
+
+  for (const q of quads) {
+    quad.p1.x = Math.min(quad.p1.x, q.p1.x);
+    quad.p1.y = Math.min(quad.p1.y, q.p1.y);
+    quad.p2.x = Math.max(quad.p2.x, q.p2.x);
+    quad.p2.y = Math.min(quad.p2.y, q.p2.y);
+    quad.p3.x = Math.max(quad.p3.x, q.p3.x);
+    quad.p3.y = Math.max(quad.p3.y, q.p3.y);
+    quad.p4.x = Math.min(quad.p4.x, q.p4.x);
+    quad.p4.y = Math.max(quad.p4.y, q.p4.y);
+
+    const qBounds = q.getBounds();
+
+    quad.bounds.bottom = Math.max(quad.bounds.bottom, qBounds.bottom);
+    quad.bounds.top = Math.min(quad.bounds.top, qBounds.top);
+    quad.bounds.left = Math.min(quad.bounds.left, qBounds.left);
+    quad.bounds.right = Math.max(quad.bounds.right, qBounds.right);
+  }
+  quad.bounds.x = quad.bounds.left;
+  quad.bounds.y = quad.bounds.top;
+  quad.bounds.width = quad.bounds.right - quad.bounds.left;
+  quad.bounds.height = quad.bounds.bottom - quad.bounds.top;
+
+  return quad;
+}
+
+export function PreviewNodeHighlighter({ nodeId }: { nodeId: string }) {
+  const elementRef = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    const element = elementRef.current;
+    if (element) {
+      return subscribe(state => {
+        const { localScale, recordingScale } = state;
+        const scale = localScale * recordingScale;
+
+        element.style.transform = `scale(${scale})`;
+        element.style.height = `calc(100% / ${scale})`;
+        element.style.width = `calc(100% / ${scale})`;
+      });
+    }
+  }, []);
```

Comment:
This is the only real change to this file. The rest of it was mostly just a `git mv` that it lost track of.